homeassistant/components/vesync/fan.py

```diff
@@ -21,8 +19,11 @@
     "LV-PUR131S": "fan",
 }
 
-FAN_SPEEDS = [SPEED_OFF, SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH]
 FAN_MODE_AUTO = "auto"
+FAN_MODE_SLEEP = "sleep"
+
+PRESET_MODES = [FAN_MODE_AUTO, FAN_MODE_SLEEP]
```

Comment:
Should there be a `FAN_MODE_NONE` so that users can disable a preset mode ?

Reply:
I'm trying to think about how I would use it.

I'm thinking I would use it like:

from `Auto` -> `Medium Speed` or `Auto` -> `Off` and wouldn't do `Auto` -> `None` -> `Medium`

Reply:
I'll do some testing with the frontend PR to make sure that is actually how I would use it with the demo fan.

Reply:
![T2Uetqxcxe](https://user-images.githubusercontent.com/663432/106155639-84e2e780-6146-11eb-9cea-72aa28889fb5.gif)

If the device still reports the fan level in `Auto` mode, maybe it makes sense to add `Manual` (suggested as `None`).

If the device doesn't report the fan level in auto mode then it probably is fine as is.

Reply:
I ordered a `LV-PUR131S` for testing

Reply:
> I ordered a `LV-PUR131S` for testing and will adjust later if needed

I am available to test this when you feel it is ready as well. I have 3 of these.

Reply:
If you could give the latest dev a try along with latest frontend with https://github.com/home-assistant/frontend/pull/8216 and provide feedback, that would be most excellent.